src/main/java/back/calculator/ExpressionParser.java

```diff
@@ -52,8 +53,9 @@ public Expression parse(String expression) throws IllegalConstruction {
 
         if (errorListener.hasSyntaxError()) {
             // Report the syntax error
-            // TODO: Log the error message here
-            System.err.println(errorListener.getMessage());
+            Logger logger = App.getLogger();
+            String errorMessage = errorListener.getMessage();
+            logger.error(errorMessage);
```

Comment:
Shouldn't we define a few logging methods in App instead of having to call App.getLogger() every time ?

Reply:
I modified it. Now, you have to create a logger in the class. With this, the name of the class will be printed in the log file and will be more readable, I think.

I left the old code in App because it may be used in the future to log something with the "App" name.

Reply:
Good idea